controllers/awsadapterconfig_controller.go

```diff
@@ -297,6 +297,47 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 					}
 				}
 
+				x, err := ec2Client.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{
+					Filters: []types.Filter{
+						{
+							Name: aws.String("tag:aws:eks:cluster-name"),
+							Values: []string{
+								*objOld.Spec.Name,
+							},
+						},
+						{
+							Name: aws.String("tag:eks:nodegroup-name"),
+							Values: []string{
+								v,
+							},
+						},
+					},
+				})
+				if err != nil {
+					l.Error(err, "error occurred while fetching EC2 instances")
+					return r.updateLastPollStatusFailure(ctx, objOld, "error occurred while fetching EC2 instances", err, &l, time.Now())
+				}
+
+				ami, err := getAmi(ctx, ec2Client, x.Reservations[0].Instances[0].ImageId)
+				if err != nil {
```

Comment:
So it seems we are saving one of the many AMIs of the instances that a nodegroup might have. Because Nodegroup has a single AMIType, but multiple AMI Ids corresponding to that type, with variations in AMI fields like location, createtime, owner etc. Depending upon what fields of AMI are of interest, we might be ok with storing any one of the AMIs.